commercetools-sdk-base/src/main/java/io/sphere/sdk/client/CorrelationIdGenerator.java

```diff
@@ -0,0 +1,31 @@
+package io.sphere.sdk.client;
+
+/**
+ * Generates correlation ids for each request.
+ *
+ * The format of the correlation id is defined as:
+ * {@code <projectKey> / <UUID> (/ customId)*}
+ */
+public interface CorrelationIdGenerator {
```

Comment:
I agree that users could write their own decorator, but still think it would be useful to expose this interface so that our users can customize the correlation id generator. And I think exposing this interface makes it easier to customize the correlation id generation. I'll think about it further and then we can discuss it.